src/Dictionaries/CacheDictionary.inc.h

```diff
@@ -176,56 +175,70 @@ void CacheDictionary::getItemsString(
 {
     const auto rows = ext::size(ids);
 
-    /// save on some allocations
+    /// Save on some allocations.
     out->getOffsets().reserve(rows);
 
     auto & attribute_array = std::get<ContainerPtrType<StringRef>>(attribute.arrays);
 
     auto found_outdated_values = false;
 
-    /// perform optimistic version, fallback to pessimistic if failed
+    /// Perform optimistic version, fallback to pessimistic if failed.
     {
         const ProfilingScopedReadRWLock read_lock{rw_lock, ProfileEvents::DictCacheLockReadNs};
 
         const auto now = std::chrono::system_clock::now();
-        /// fetch up-to-date values, discard on fail
+        /// Fetch up-to-date values, discard on fail.
         for (const auto row : ext::range(0, rows))
         {
             const auto id = ids[row];
 
-            const auto find_result = findCellIdx(id, now);
+            const auto find_result = findCellIdxForGet(id, now);
+            auto & cell = cells[find_result.cell_idx];
+
+            auto insert_routine = [&] ()
+            {
+                const auto & cell_idx = find_result.cell_idx;
+                const auto string_ref = cell.isDefault() ? get_default(row) : attribute_array[cell_idx];
+                out->insertData(string_ref.data, string_ref.size);
```

Comment:
Why is it a lambda? Two lines with `const auto` can go before `if`, and you can just call `insertData` inside the branches.

Reply:
Fixed.